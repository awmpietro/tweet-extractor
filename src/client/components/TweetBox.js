import React from 'react';
import PropTypes from 'prop-types';
import { Col } from 'reactstrap';

const TweetBox = ({ result }) => {
  return (
    <Col md="4" className="bootstrap snippets">
      <div className="widget panel">
        <div className="panel-body">
          <ul className="list-table">
            <li style={{ width: 60 }}>
              <img
                className="img-circle mb-2"
                src={result.user.profile_image_url}
                alt=""
                width="55"
                height="55"
              />
            </li>
            <li className="text-left">
              <h4 className="semibold ellipsis nm">
                {result.user.screen_name}
              </h4>
              <p className="text-muted nm">
                {result.user.followers_count} followers
              </p>
            </li>
            <li className="text-right">
              <button
                type="button"
                className="btn btn-sm btn-info"
                onClick={() =>
                  window.open(
                    `https://twitter.com/i/web/status/${result.id_str}`,
                    '_blank',
                  )
                }
              >
                See
              </button>
            </li>
          </ul>
        </div>

        <ul className="list-group">
          <li className="list-group-item">
            <p className="nm">{result.text}</p>
            <small className="text-muted">{result.created_at}</small>
          </li>
        </ul>
      </div>
    </Col>
  );
};

TweetBox.propTypes = {
  result: PropTypes.object,
};

export default TweetBox;
