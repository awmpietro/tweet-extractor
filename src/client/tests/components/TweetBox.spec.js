import React from 'react';
import TweetBox from '../../components/TweetBox';
import renderer from 'react-test-renderer';

describe('TweetBox', () => {
  test('snapshot renders', () => {
    const result = {"user" : {}};
    const component = renderer.create(<TweetBox result={result} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});