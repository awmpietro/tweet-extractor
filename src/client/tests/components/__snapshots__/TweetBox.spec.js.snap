// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetBox snapshot renders 1`] = `
<div
  className="bootstrap snippets col-md-4"
>
  <div
    className="widget panel"
  >
    <div
      className="panel-body"
    >
      <ul
        className="list-table"
      >
        <li
          style={
            Object {
              "width": 60,
            }
          }
        >
          <img
            alt=""
            className="img-circle mb-2"
            height="55"
            width="55"
          />
        </li>
        <li
          className="text-left"
        >
          <h4
            className="semibold ellipsis nm"
          />
          <p
            className="text-muted nm"
          >
             followers
          </p>
        </li>
        <li
          className="text-right"
        >
          <button
            className="btn btn-sm btn-info"
            onClick={[Function]}
            type="button"
          >
            See
          </button>
        </li>
      </ul>
    </div>
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      >
        <p
          className="nm"
        />
        <small
          className="text-muted"
        />
      </li>
    </ul>
  </div>
</div>
`;
